cmd/src/headers.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"os"
+	"strings"
+)
+
+// parseAdditionalHeaders reads the environment for values like SRC_HEADER_NAME=VALUE
+// and creates a `{NAME: VALUE}` map. These headers should be applied to each request
+// to the Sourcegraph instance, as some private instances require special auth or
+// additional proxy values to be passed along with each request.
+func parseAdditionalHeaders() map[string]string {
+	return parseAdditionalHeadersFromMap(os.Environ())
+}
+
+const additionalHeaderPrefix = "SRC_HEADER_"
+
+func parseAdditionalHeadersFromMap(environ []string) map[string]string {
+	additionalHeaders := map[string]string{}
+	for _, value := range environ {
+		parts := strings.SplitN(value, "=", 2)
+		if len(parts) != 2 || len(parts[0]) <= len(additionalHeaderPrefix) || len(parts[1]) == 0 || !strings.HasPrefix(parts[0], additionalHeaderPrefix) {
```

Comment:
You don't need to do ```len(parts[0]) <= len(additionalHeaderPrefix)```, strings.HasPrefix does the same check

Reply:
This checks that it’s a proper prefix and it’s not the exact string so that when we trim below there’s still something left.

Reply:
Adding additional docs here.